File: 0075 sort-colors/solution.rs
impl Solution {
    pub fn sort_colors(nums: &mut Vec<i32>) {
        let mut low:usize = 0;
        let mut mid:usize = 0;
        let mut high:usize = nums.len()-1;

        while mid<=high {
            if nums[mid]==0 {
                nums.swap(mid,low);
                mid+=1;
                low+=1;
            } else if nums[mid] == 1 {
                mid+=1;
            } else {
                nums.swap(mid,high);
                if high==0 {
                    break;
                }
                high-=1;
            }
        }
    }
}

File: 0075 sort-colors/solution.cpp
class Solution {
public:
    void sortColors(vector<int>& nums) {
        int z = 0, o = 0, t = nums.size()-1;
        while(o<=t) {
            if(nums[o]==0) {
                nums[o] = nums[z];
                nums[z] = 0;
                z++;
                o++;
            } else if(nums[o]==1) {
                o++;
            } else {
                nums[o] = nums[t];
                nums[t] = 2;
                t--;
            }
        }
    }
};
